From 0cef6cd063bf19f776ddf75e7bda7fc3ab5efec3 Mon Sep 17 00:00:00 2001
From: Christophe Roullier <christophe.roullier@foss.st.com>
Date: Tue, 28 May 2024 11:49:23 +0200
Subject: [PATCH] arm64: dts: st: add Ethernet1 pinmux nodes in
 stm32mp21-pinctrl.dtsi

Add eth1 pinmux nodes in stm32mp21-pinctrl.dtsi

Change-Id: I2f3235dc615d3acbb4fcfe165d3016ca2b0e937c
Signed-off-by: Christophe Roullier <christophe.roullier@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/linux-stm32/+/383313
Reviewed-by: Gwenael TREUVEUR <gwenael.treuveur@st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 arch/arm64/boot/dts/st/stm32mp21-pinctrl.dtsi | 36 +++++++++++++++++++
 1 file changed, 36 insertions(+)

diff --git a/arch/arm64/boot/dts/st/stm32mp21-pinctrl.dtsi b/arch/arm64/boot/dts/st/stm32mp21-pinctrl.dtsi
index 03c850cb51bc..698e1ebd7654 100644
--- a/arch/arm64/boot/dts/st/stm32mp21-pinctrl.dtsi
+++ b/arch/arm64/boot/dts/st/stm32mp21-pinctrl.dtsi
@@ -6,6 +6,42 @@
 #include <dt-bindings/pinctrl/stm32-pinfunc.h>
 
 &pinctrl {
+	eth1_rmii_pins_a: eth1-rmii-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX('A', 15, AF10)>, /* ETH_RMII_TXD0 */
+				 <STM32_PINMUX('C', 1, AF10)>, /* ETH_RMII_TXD1 */
+				 <STM32_PINMUX('A', 13, AF10)>, /* ETH_RMII_TX_EN */
+				 <STM32_PINMUX('F', 8, AF9)>, /* ETH_RMII_ETHCK */
+				 <STM32_PINMUX('F', 2, AF10)>, /* ETH_MDIO */
+				 <STM32_PINMUX('F', 0, AF10)>; /* ETH_MDC */
+			bias-disable;
+			drive-push-pull;
+			slew-rate = <1>;
+		};
+
+		pins2 {
+			pinmux = <STM32_PINMUX('F', 1, AF10)>, /* ETH_RMII_RXD0 */
+				 <STM32_PINMUX('C', 2, AF10)>, /* ETH_RMII_RXD1 */
+				 <STM32_PINMUX('A', 11, AF10)>; /* ETH_RMII_CRS_DV */
+			bias-disable;
+		};
+
+	};
+
+	eth1_rmii_sleep_pins_a: eth1-rmii-sleep-0 {
+		pins1 {
+			pinmux = <STM32_PINMUX('A', 15, ANALOG)>, /* ETH_RMII_TXD0 */
+				 <STM32_PINMUX('C', 1, ANALOG)>, /* ETH_RMII_TXD1 */
+				 <STM32_PINMUX('A', 13, ANALOG)>, /* ETH_RMII_TX_EN */
+				 <STM32_PINMUX('F', 8, ANALOG)>, /* ETH_RMII_ETHCK */
+				 <STM32_PINMUX('F', 2, ANALOG)>, /* ETH_MDIO */
+				 <STM32_PINMUX('F', 0, ANALOG)>, /* ETH_MDC */
+				 <STM32_PINMUX('F', 1, ANALOG)>, /* ETH_RMII_RXD0 */
+				 <STM32_PINMUX('C', 2, ANALOG)>, /* ETH_RMII_RXD1 */
+				 <STM32_PINMUX('A', 11, ANALOG)>; /* ETH_RMII_CRS_DV */
+		};
+	};
+
 	goodix_pins_a: goodix-0 {
 		/*
 		 * touchscreen reset needs to be configured
-- 
2.39.5

